src/main/java/com/bootcamp/demo/service/ScooterServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+import com.bootcamp.demo.model.util.Location;
+import com.bootcamp.demo.service.assembler.ScooterAssembler;
+import com.bootcamp.demo.service.exception.ServiceException;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+@Slf4j
+public class ScooterServiceImpl implements ScooterService{
+
+    private final Firestore db;
+    private static final String COLLECTION_SCOOTERS_PATH = "bookings/databases/scooters";
+    private static final ResponseEntity<Object> SUCCESS_RESPONSE = new ResponseEntity<>(HttpStatus.OK);
+    private static final ResponseEntity<Object> FAILURE_RESPONSE = new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+
+    private ScooterServiceImpl(Firestore db) {
+        this.db = db;
+    }
+
+    @Override
+    public ResponseEntity<Scooter> findScooterById(String id) {
+        LOGGER.info("FIND SCOOTER BY ID - service function invoked");
+        return null;
+    }
+
+    @Override
+    public Set<Scooter> findAllScooters() {
+        LOGGER.info("FIND ALL SCOOTER - service function invoked");
+
+        try {
+            ApiFuture<QuerySnapshot> querySnapshotApiFuture  = db.collection(COLLECTION_SCOOTERS_PATH).get();
+
+            Iterable<QueryDocumentSnapshot> scootersDocuments = querySnapshotApiFuture.get().getDocuments();
+
+            LOGGER.info("Found scooters successfully");
+
+            return StreamSupport
+                    .stream(scootersDocuments.spliterator(), false)
+                    .map(ScooterAssembler::documentToModel)
+                    .collect(Collectors.toSet());
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error(e.getMessage());
+            throw new ServiceException();
+        }
+
+    }
+
+    @Override
+    public void createScooter(final Scooter scooter) {
+        LOGGER.info("CREATE SCOOTER - service function invoked");
+
+        try {
+            ApiFuture<WriteResult> collectionsApiFuture = db.collection(COLLECTION_SCOOTERS_PATH)
+                    .document(scooter.getSerialNumber())
+                    .set(scooter);
+
+            String update_time = collectionsApiFuture.get().getUpdateTime().toDate().toString();
+
+            LOGGER.info("SCOOTER create successfully. Update_time: " + update_time);
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error(e.getMessage());
+            throw new ServiceException();
+        }
+    }
+
+    private boolean approximateLocation(Location userLocation, Location scooterLocation) {
+        if (scooterLocation.getLongitude() <= userLocation.getLongitude() + 0.01
```

Comment:
Since the conditions from if are very similar, wondering if we could extract this to another method which would fo the comparison for a coordinate & apply it for both. This would result in sth like isNearby(user long/lat, scooter long/lat), which would be called for both. 
If that will be done, you could also use a ternary for this.